examples/01_icetray/05_convert_i3_files_advanced.py

```diff
@@ -0,0 +1,179 @@
+"""More advanced example script to convert IceCube I3 files to sqlite."""
+
+from glob import glob
+
+from graphnet.constants import EXAMPLE_OUTPUT_DIR, TEST_DATA_DIR
+from graphnet.data.extractors.icecube import (
+    I3TruthExtractor,
+    I3ParticleExtractor,
+    I3FeatureExtractorIceCube86,
+    I3GenericExtractor,
+    I3DictValueExtractor,
+    I3FilterMapExtractor,
+    I3HighestEparticleExtractor,
+    I3Calorimetry,
+)
+
+from graphnet.data.extractors.icecube.utilities.i3_filters import (
+    TableFilter,
+    NullSplitI3Filter,
+    ChargeFilter,
+)
+
+from graphnet.data.extractors.icecube.utilities.gcd_hull import GCD_hull
+
+from graphnet.data.extractors.combine_extractors import CombinedExtractor
+
+
+from graphnet.data.dataconverter import DataConverter
+from graphnet.data.pre_configured import I3ToSQLiteConverter
+from graphnet.utilities.argparse import ArgumentParser
+from graphnet.utilities.imports import has_icecube_package
+from graphnet.utilities.logging import Logger
+
+ERROR_MESSAGE_MISSING_ICETRAY = (
+    "This example requires IceTray to be installed, which doesn't seem to be "
+    "the case. Please install IceTray; run this example in the GraphNeT "
+    "Docker container which comes with IceTray installed; or run an example "
+    "script in one of the other folders:"
+    "\n * examples/02_data/"
+    "\n * examples/03_weights/"
+    "\n * examples/04_training/"
+    "\n * examples/05_pisa/"
+    "\nExiting."
+)
+
+
+def main(
+    merge: bool,
+    remove: bool,
+    workers: int = 1,
+    padding: int = 250,
+    max_table_size: int = int(25e7),
+) -> None:
+    """Convert IceCube-Upgrade I3 files to intermediate `backend` format."""
+    # Check(s)
+    inputs = [f"{TEST_DATA_DIR}/i3/oscNext_muongun_level3_v02"]
```

Comment:
Is this test file sufficient for the example, or should be add a new one?

Reply:
The current file does produce any events meaning a db is not being constructed, it would be nice if we could add a file nugen file with an event passing the charge cut (charge cut could also be reduced).